src/expr.mli

```diff
@@ -176,83 +178,190 @@ module T : sig
-  val adjs : ctx -> Util.hint list -> ctx * string list
-  val fmt_bounds :
-    ctx -> bound list -> ctx * (Format.formatter -> unit)
-  val extend_bounds :
-    ?prevdom:expr option ->
-    ctx -> bound list -> ctx * (string * expr option) list
-  val pp_print_shape : Format.formatter -> shape -> unit
-  val pp_print_bound :
-    ctx -> Format.formatter -> string * expr option -> unit
-  val fmt_expr : ?temp:bool -> ctx -> expr -> Tla_parser.Fu.exp
-  val pp_print_expr : ?temp:bool -> ctx -> Format.formatter -> expr -> unit
-  val pp_print_defn : ctx -> Format.formatter -> defn -> ctx
-  val pp_print_defns : ctx -> Format.formatter -> defn list -> ctx
-  val pp_print_sequent : ?temp:bool -> ctx -> Format.formatter -> sequent -> ctx
-  val pp_print_hyp : ?temp:bool -> ctx -> Format.formatter -> hyp -> ctx
-  val pp_print_instance : ctx -> Format.formatter -> instance -> unit
-  val string_of_expr : hyp Deque.dq  -> expr -> string
-end;;
-
-module Subst : sig
-  open Property
-  open T
-  type sub
-  val shift : int -> sub
-  val scons : expr -> sub -> sub
-  val bumpn : int -> sub -> sub
-  val bump : sub -> sub
-  val compose : sub -> sub -> sub
-  val app_ix : sub -> int wrapped -> expr
-  val normalize : ?cx:hyp Deque.dq -> expr -> expr list -> expr_
-  val app_expr : sub -> expr -> expr
-  val app_bdom : sub -> bound_domain -> bound_domain
-  val app_bound : sub -> bound -> sub * bound
-  val app_bounds : sub -> bound list -> sub * bound list
-  val app_defn : sub -> defn -> defn
-  val app_defns : sub -> defn list -> sub * defn list
-  val app_sequent : sub -> sequent -> sequent
-  val app_hyp : sub -> hyp -> hyp
-
-  class map : object
-      method expr     : sub -> expr -> expr
-      method exprs    : sub -> expr list -> expr list
-      method pform    : sub -> pform -> pform
-      method sels     : sub -> sel list -> sel list
-      method sel      : sub -> sel -> sel
-      method sequent  : sub -> sequent -> sub * sequent
-      method defn     : sub -> defn -> sub * defn
-      method defns    : sub -> defn list -> sub * defn list
-      method bounds   : sub -> bound list -> sub * bound list
-      (* TODO: method bound    : sub -> bound -> sub * bound *)
-      method bound    : sub -> bound -> bound
-      method exspec   : sub -> exspec -> exspec
-      method instance : sub -> instance -> instance
-      method hyp      : sub -> hyp -> sub * hyp
-      method hyps     : sub -> hyp Deque.dq -> sub * hyp Deque.dq
-      method normalize : ?cx:hyp Deque.dq -> expr -> expr list -> expr_
-  end
 
-  class map_visible_hyp : map
-end;;
+module Fmt: sig
+    open T
+    open Ctx
+    type ctx = hyp Deque.dq * int Ctx.ctx
+    val bump:
+        ctx -> ctx
+    val adj:
+        ctx -> Util.hint ->
+            ctx * string
+    val adjs:
+        ctx -> Util.hint list ->
+            ctx * string list
+    val is_letter:
+        char -> bool
+    val is_hidden:
+        hyp -> bool
+    val elide:
+        hyp -> bool
```

Comment:
Thank you for noting this, I moved these new declarations to another pull request, because they are refactoring that was motivated by changes related to tuply declarations.